From 8abd76884d8be825c4e4dd9cf251a02ced1eca2b Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Wed, 4 Mar 2026 01:04:33 +0000
Subject: [PATCH] CMakeLists: prevent installing html docs

Prevent the installing html docs when BUILDS_DOC is disabled

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 CMakeLists.txt | 49 ++++++++++++++++++++++++-------------------------
 1 file changed, 24 insertions(+), 25 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f92ecb0..c9047b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,36 +49,35 @@ install(FILES "${GENERATED_CONFIG_BUILD_PATH}"
   COMPONENT lib
 )
 
-
-if(DOC_INSTALL_DIR)
-else()
-  if(WIN32)
-    set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/doc)
-  else()
-    set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/tclap)
-  endif()
-endif()
-
-cpack_add_component(doc
-	DISPLAY_NAME "documentation"
-	DESCRIPTION "Documentation files for TCLAP"
-)
-
-install(DIRECTORY ${PROJECT_BINARY_DIR}/docs/html
-  DESTINATION ${DOC_INSTALL_DIR}
-  COMPONENT doc
-)
-install(FILES ${PROJECT_BINARY_DIR}/docs/index.html ${PROJECT_BINARY_DIR}/docs/manual.html
-  ${PROJECT_BINARY_DIR}/docs/style.css DESTINATION ${DOC_INSTALL_DIR}
-  COMPONENT doc
-)
-
 option(BUILD_EXAMPLES "Build examples" ON)
 option(BUILD_UNITTESTS "Build unit tests" ON)
 option(BUILD_DOC "Build documentation" ON)
 
 if(BUILD_DOC)
-	add_subdirectory(docs)
+  if(DOC_INSTALL_DIR)
+  else()
+    if(WIN32)
+      set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/doc)
+    else()
+      set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/tclap)
+    endif()
+  endif()
+
+  cpack_add_component(doc
+    DISPLAY_NAME "documentation"
+    DESCRIPTION "Documentation files for TCLAP"
+  )
+
+  install(DIRECTORY ${PROJECT_BINARY_DIR}/docs/html
+    DESTINATION ${DOC_INSTALL_DIR}
+    COMPONENT doc
+  )
+  install(FILES ${PROJECT_BINARY_DIR}/docs/index.html ${PROJECT_BINARY_DIR}/docs/manual.html
+    ${PROJECT_BINARY_DIR}/docs/style.css DESTINATION ${DOC_INSTALL_DIR}
+    COMPONENT doc
+  )
+
+  add_subdirectory(docs)
 endif()
 if(BUILD_EXAMPLES)
 	add_subdirectory(examples)
-- 
2.52.0

